python/lsst/ts/dsm/dsm_csc.py

```diff
@@ -233,6 +233,5 @@ async def simulated_telemetry_loop(self):
 
     async def telemetry_loop(self):
         """Run the telemetry loop."""
-        while True:
-            ioevent = await self.telemetry_watcher.get_event()
+        async for ioevent in self.telemetry_notifier:
```

Comment:
Since you only use the telemetry notifier in this loop, please consider just putting all the watcher code here, e.g.:
```
async def telemetry_loop(self):
    with asyncinotify.Inotify() as inotify:
        inotify.add_watch(path=self.telemetry_directory, mask=asyncinotify.Mask.CLOSE_WRITE)
        async for ioevent in self.telemetry_notifier:
            self.process_event(ioevent)
```
The advantages are:
- You don't have to store the notifier or watch in instance variables; all knowledge is local
- You don't have to clean up the notifier or watch when you're finished; the context managers do it for you
- Your handle_summary_state only has to start or stop the telemetry loop task; it does not have to add a watch. The block of code that creates the watch can just be deleted.
- You avoid any attempt to add a watch to a running notifier -- it's not even clear to me what that should do, but it definitely should not be needed.
- You eliminate the need for the new code in close_tasks.